internal/tiltfile/tiltfile_state.go

```diff
@@ -1692,6 +1678,53 @@ func toKubernetesApplyCmd(cmd model.Cmd) *v1alpha1.KubernetesApplyCmd {
 	}
 }
 
+func (s *tiltfileState) ignoresForImage(image *dockerImage) (contextIgnores []v1alpha1.IgnoreDef, fileWatchIgnores []v1alpha1.IgnoreDef, err error) {
+	dockerignores, err := s.dockerignoresForImage(image)
+	if err != nil {
+		return nil, nil, fmt.Errorf("reading dockerignore for %s: %v", image.configurationRef.RefFamiliarString(), err)
+	}
+	if image.tiltfilePath != "" {
+		// TODO(milas): technically, this should be a FW ignore, not a context
+		// 	ignore, but historically it has been this way
+		contextIgnores = append(contextIgnores, v1alpha1.IgnoreDef{BasePath: image.tiltfilePath})
+	}
+	contextIgnores = append(contextIgnores, s.repoIgnoresForImage(image)...)
+	contextIgnores = append(contextIgnores, model.DockerignoresToIgnores(dockerignores)...)
+
+	for i := range contextIgnores {
+		fileWatchIgnores = append(fileWatchIgnores, *contextIgnores[i].DeepCopy())
```

Comment:
fyi - i think the k8s api convention is to only do a copy when you know you're going to modify the object

Reply:
Yeah, this is probably extra defensive but given the lifetime of these items and how much they get passed around (+ the fact that we've got two intersecting sets of them originating from the same objects), I figured an extra copy was worth saving potential future debugging headache down the road